pkg/stanza/operator/parser/keyvalue/parser_test.go

```diff
@@ -688,6 +688,24 @@ key=value`,
 			true,
 			false,
 		},
```

Comment:
some edge cases that may be helpful for fully defining expected behavior:
- what do you expect the behavior to be when there is only a single quote? ie an incomplete escape sequence, a quote but no unquote ie `time="2024 level=warning`
- what about nesting? `msg="warning message 'the input was "level"'"`